File: Frontend/Certiva/src/app/Models/usuario-dto.ts
export interface UsuarioDTO {
  idUsuario: number;
  nombres: string;
  apellidos: string;
  numeroDocumento: string;
  correo: string;
  telefono?: string;
  estado: boolean | string;
  fechaRegistro: string;
  idRol?: number;
  idTipoDocumento?: number;
  rol?: {
    idRol: number;
    nombre: string;
    codigo?: string;
  };
  tipoDocumento?: {
    idTipoDocumento: number;
    nombre: string;
  };
}

File: Backend/api/src/main/java/com/certiva/api/DTO/UsuarioDTO.java
package com.certiva.api.DTO;

import java.time.LocalDateTime;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UsuarioDTO {

    private Long idUsuario;

    @NotBlank(message = "Los nombres son obligatorios")
    @Size(max = 100)
    private String nombres;

    @NotBlank(message = "Los apellidos son obligatorios")
    @Size(max = 100)
    private String apellidos;

    @NotBlank(message = "El número de documento es obligatorio")
    @Size(max = 50)
    private String numeroDocumento;

    @NotBlank(message = "El correo es obligatorio")
    @Email(message = "El correo debe tener un formato válido")
    @Size(max = 150)
    private String correo;

    @Size(max = 20, message = "El teléfono no puede superar 20 caracteres")
    private String telefono;

    private Boolean estado;

    private LocalDateTime fechaRegistro;

    @NotNull(message = "El tipo de documento es obligatorio")
    private Long idTipoDocumento;

    private Long idRol;

    private RolResumenDTO rol;

    private TipoDocumentoResumenDTO tipoDocumento;
}
